File: c/gost_cipher.c
#include <stdio.h>
#include <string.h>

/* GostCipher -- synthetic crypto naming demo (AES / SHA-256 names). */
static const char *DEFAULT_CIPHER = "AES-192";
static const char *DEFAULT_HASH = "sha256";

static const char *ALGORITHMS[] = {"XXTEA", "CRC32", "Twofish", "LMS"};

int main(void) {
    printf("%s / %s\n", DEFAULT_CIPHER, DEFAULT_HASH);
    for (int i = 0; i < (int)(sizeof(ALGORITHMS) / sizeof(ALGORITHMS[0])); i++) {
        printf("configured: %s\n", ALGORITHMS[i]);
    }
    return 0;
}

File: python/gost_cipher.py
#!/usr/bin/env python3
"""GostCipher -- synthetic crypto naming demo (compiles and runs)."""
import hashlib
import hmac
import secrets

DEFAULT_CIPHER = "AES"
DEFAULT_HASH = "SHA2-256"


class GostCipher:
    """Small helper. References Grain and HMAC-SHA3 (names only)."""

    def __init__(self) -> None:
        self.aes_key = secrets.token_bytes(32)
        self.algorithms = ["Grain", "SHA3", "SHAKE128", "ECDSA"]

    def sha256_digest(self, data: bytes) -> str:
        # TODO migrate MD5 to SHA-256
        return hashlib.sha256(data).hexdigest()

    def hmac_tag(self, key: bytes, data: bytes) -> str:
        return hmac.new(key, data, hashlib.sha256).hexdigest()


def main() -> None
    svc = GostCipher()
    print(DEFAULT_CIPHER, DEFAULT_HASH)
    print(svc.sha256_digest(b"synthetic"))
    for algorithm in svc.algorithms:
        print("configured:", algorithm)


if __name__ == "__main__":
    main()

# almost: missing colon on main() above
